bigtable_rs/src/bigtable.rs

```diff
@@ -248,53 +249,80 @@ impl BigTableConnection {
-                                    )
-                                    .domain_name("bigtable.googleapis.com"),
-                            )
-                            .map_err(Error::TransportError)
-                    })
-                    .collect();
-
-                let endpoints: Vec<Endpoint> = endpoints?
-                    .into_iter()
-                    .map(|ep| {
-                        ep.http2_keep_alive_interval(Duration::from_secs(60))
-                            .keep_alive_while_idle(true)
-                    })
-                    .map(|ep| {
-                        if let Some(timeout) = timeout {
-                            ep.timeout(timeout)
-                        } else {
-                            ep
-                        }
-                    })
-                    .collect();
-
-                let auth_manager = Some(Arc::new(authentication_manager));
-                Ok(Self {
-                    client: create_client(endpoints, auth_manager, is_read_only),
-                    table_prefix: Arc::new(table_prefix),
-                    timeout: Arc::new(timeout),
-                })
+                Self::new_with_auth_manager(project_id, instance_name, is_read_only, channel_size, timeout, authentication_manager).await
             }
         }
     }
+    /// Establish a connection to the BigTable instance named `instance_name`.  If read-only access
+    /// is required, the `read_only` flag should be used to reduce the requested OAuth2 scope.
+    ///
+    /// The `authentication_manager` variable will be used to determine the
+    /// program name that contains the BigTable instance in addition to access credentials.
+    ///
+    ///
+    /// `channel_size` defines the number of connections (or channels) established to Bigtable
+    /// service, and the requests are load balanced onto all the channels. You must therefore
+    /// make sure all of these connections are open when a new request is to be sent.
+    /// Idle connections are automatically closed in "a few minutes". Therefore it is important to
+    /// make sure you have a high enough QPS to send at least one request through all the
+    /// connections (in every service host) every minute. If not, you should consider decreasing the
+    /// channel size. If you are not sure what value to pick and your load is low, just start with 1.
+    /// The recommended value could be 2 x the thread count in your tokio environment see info here
+    /// https://docs.rs/tokio/latest/tokio/attr.main.html, but it might be a very different case for
+    /// different applications.
+    ///
+    pub async fn new_with_auth_manager(
+        project_id: &str,
+        instance_name: &str,
+        is_read_only: bool,
+        channel_size: usize,
+        timeout: Option<Duration>,
+        authentication_manager: AuthenticationManager,
+    ) -> Result<Self> {        
```

Comment:
I think we should move the whole inner block of the old `new()` method here (for example including the whole `match std::env::var("BIGTABLE_EMULATOR_HOST") {...}` block here), so that the emulator connect can always work no matter how one is creating a client.

To make the Git diff look smaller, I think you can just change on the exactly old line of `new()` to `new_with_auth_manager()`, then creates a small `new()` in then end of `new_with_auth_manager()`.

Let me know if it makes any sense :)

Reply:
Is the shifting of `match std::env::var("BIGTABLE_EMULATOR_HOST") {...}` correct?

Reply:
Yep, I think it looks good now.